apps/tlon-mobile/src/screens/ChannelScreen.tsx

```diff
@@ -64,27 +75,98 @@ export default function ChannelScreen(props: ChannelScreenProps) {
   const { bottom } = useSafeAreaInsets();
   const { ship } = useShip();
 
-  const messageSender = async (content: JSONContent, channelId: string) => {
-    if (!ship || !channel) {
-      return;
-    }
+  const resetImageAttachment = useCallback(() => {
+    setImageAttachment(null);
+    setUploadedImage(null);
+    uploader?.clear();
+  }, [uploader]);
 
-    const channelType = channel.type;
+  const messageSender = useCallback(
+    async (content: JSONContent, channelId: string, blocks: ub.Block[]) => {
+      if (!ship || !channel) {
+        return;
+      }
 
-    if (channelType === 'dm' || channelType === 'groupDm') {
-      await sendDirectMessage(channelId, content, ship);
-      return;
-    }
+      const channelType = channel.type;
 
-    await sendPost(channelId, content, ship);
-  };
+      if (channelType === 'dm' || channelType === 'groupDm') {
+        await sendDirectMessage(channelId, content, ship, blocks);
+        resetImageAttachment();
+        return;
+      }
+
+      await sendPost(channelId, content, ship, blocks);
+      resetImageAttachment();
+    },
+    [ship, channel, resetImageAttachment]
+  );
 
   useEffect(() => {
     if (error) {
       console.error(error);
     }
   }, [error]);
 
+  const fetchImageFromUri = useCallback(
```

Comment:
Can we pull out most or all of the image upload logic here? Some of it (like this method) could be part of `storage`, some could be hooks. We'll end up needing image upload in a number of places.